### PR TITLE
feat: add options param to query#runStream

### PR DESCRIPTION
**Pull request by AVaksman**
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

**Comment by stephenplusplus** [comment]
@AVaksman could you please explain the motivation of this PR and more about the changes?

**Comment by AVaksman** [comment]
> @AVaksman could you please explain the motivation of this PR and more about the changes?

Sure 
`query.runStream()` delegates to `request.runQueryStream()` and typescript users don't have an ability to pass `options`
```ts
export interface RunQueryOptions {
  consistency?: 'strong' | 'eventual';
  gaxOptions?: CallOptions;
  wrapNumbers?: boolean | IntegerTypeCastOptions;
}
```
in particular the `wrapNumbers` value, leaving the user no way to handle out of bounds integers.

File: system-test/datastore.ts [hunk 1]
```diff
@@ -619,6 +619,32 @@ describe('Datastore', () => {
...
+
+    it('should run a transaction query as a stream via query#runStream', done => {
+      const transaction = datastore.transaction();
+      const q = transaction.createQuery('Character').hasAncestor(ancestor);
+      let resultsReturned = 0;
+      q.runStream()
+        .on('error', done)
+        .on('data', () => resultsReturned++)
+        .on('end', () => {
+          console.log('total results are => ' + resultsReturned);
```
Comment (stephenplusplus) [inline system-test/datastore.ts]: I don't think we should log from the tests, just keeping the output clean 👍

**Comment by stephenplusplus** [comment]
Ah, gothca. Thanks! I just had a small change request, but LGTM.

File: system-test/datastore.ts (hunk 1 above)
Comment (AVaksman) [reply system-test/datastore.ts]: Done! 
Thanks

Anything else jumps out?

Comment (stephenplusplus) [reply system-test/datastore.ts]: Nope, LGTM. Thanks!

**Pull request merged**